File: app/src/main/java/xyz/rodit/snapmod/features/tweaks/DisableBitmojis.java
package xyz.rodit.snapmod.features.tweaks;

import de.robv.android.xposed.XC_MethodHook;
import xyz.rodit.snapmod.features.Feature;
import xyz.rodit.snapmod.features.FeatureContext;
import xyz.rodit.snapmod.mappings.BitmojiUriHandler;

public class DisableBitmojis extends Feature {

    public DisableBitmojis(FeatureContext context) {
        super(context);
    }

    @Override
    protected void performHooks() {
        // Disable Bitmoji avatars.
        BitmojiUriHandler.handle.hook(new XC_MethodHook() {
            @Override
            protected void beforeHookedMethod(MethodHookParam param) {
                if (context.config.getBoolean("disable_bitmojis")) {
                    param.setResult(null);
                }
            }
        });
    }
}

File: app/src/main/java/xyz/rodit/snapmod/features/tweaks/DisableBitmojis.kt
package xyz.rodit.snapmod.features.tweaks

import xyz.rodit.snapmod.features.Feature
import xyz.rodit.snapmod.features.FeatureContext
import xyz.rodit.snapmod.mappings.BitmojiUriHandler
import xyz.rodit.snapmod.util.before

class DisableBitmojis(context: FeatureContext) : Feature(context) {

    override fun performHooks() {
        // Disable Bitmoji avatars.
        BitmojiUriHandler.handle.before(context, "disable_bitmojis") { it.result = null }
    }
}